File: MultiLanguageSandbox/src/test/deps/perl/45.pl

sub calculate_arithmetic_sequence_nth_term {
    my ($a1, $a2, $n) = @_;
    
    # Calculate the common difference d
    my $d = $a2 - $a1;
    
    # Calculate the nth term using the formula: a_n = a1 + (n-1)*d
    my $nth_term = $a1 + ($n - 1) * $d;
    
    return $nth_term;
}

# Example usage:
# print calculate_arithmetic_sequence_nth_term(1, 4, 100);  # Output: 298


sub check {
    my $func = shift;
    die "Test failed! Expected 298, got " . &$func(1, 4, 100) unless &$func(1, 4, 100) == 298;
    die "Test failed! Expected 8, got " . &$func(2, 5, 3) unless &$func(2, 5, 3) == 8;
    die "Test failed! Expected 11, got " . &$func(-1, 3, 5) unless &$func(-1, 3, 5) == 15;
    die "Test failed! Expected 200, got " . &$func(1, 3, 100) unless &$func(1, 3, 100) == 199;
    die "Test failed! Expected -97, got " . &$func(-100, -99, 4) unless &$func(-100, -99, 4) == -97;
}

check(\&calculate_arithmetic_sequence_nth_term);

File: MultiLanguageSandbox/src/test/deps/haskell/45.hs
nthTerm :: Int -> Int -> Int -> Int
nthTerm a1 a2 n = a1 + (n - 1) * d
  where d = a2 - a1
check :: (Int -> Int -> Int -> Int) -> IO ()
check nthTermFunc = do
  let testCases = 
        [ ((1, 4, 100), 298),
          ((5, 7, 10), 23),
          ((-3, 1, 5), 13),
          ((0, 2, 50), 98)
        ]

  mapM_ (\((a1, a2, n), expected) -> testCase a1 a2 n expected) testCases
  where
    testCase a1 a2 n expected = do
      let result = nthTermFunc a1 a2 n
      putStrLn $
        "Input: a1=" ++ show a1 ++ ", a2=" ++ show a2 ++ ", n=" ++ show n ++
        ", Expected: " ++ show expected ++ ", Result: " ++ show result ++
        if result == expected then " (Pass)" else error "Test failed. Exiting..."

main :: IO ()
main = check nthTerm